modules/polygon/src/cut-by-mercator-world.js

```diff
@@ -0,0 +1,155 @@
+    // The insertion point is defined by the vertex closest to the pole
+    // Split the the ring by the insertion point when copying to newPositions
+    const splitIndex = findSplitIndex(positions, size, srcStartIndex, srcEndIndex);
+    for (let i = splitIndex; i < srcEndIndex; i++) {
+      newPositions[targetIndex++] = positions[i];
+    }
+    for (let i = srcStartIndex; i < splitIndex; i++) {
+      newPositions[targetIndex++] = positions[i];
+    }
+
+    // Remap longitudes so that each segment takes the shorter path
+    wrapLongitudesForShortestPath(newPositions, size, targetStartIndex, targetIndex);
+
+    // Handle the case when the ring contains a pole
+    insertPoleVertices(newPositions, size, targetStartIndex, targetIndex);
+
+    srcStartIndex = srcEndIndex;
+    newHoleIndices[ringIndex] = targetIndex;
+  }
+  newHoleIndices.pop();
+
+  const parts = cutPolygonByGrid(newPositions, newHoleIndices, {
+    size,
+    gridResolution: GRID_RESOLUTION,
+    gridOffset: GRID_OFFSET
+  });
+
+  if (normalize) {
+    // Each part is guaranteed to be in a single copy of the world
+    // Map longitudes back to [-180, 180]
+    for (const part of parts) {
+      shiftLongitudesIntoRange(part.positions || part, size);
+    }
+  }
+  return parts;
+}
+
+/* Helpers */
+
+function findSplitIndex(positions, size, startIndex, endIndex) {
+  let maxLat = -1;
+  let pointIndex = -1;
+  for (let i = startIndex + 1; i < endIndex; i += size) {
+    const lat = Math.abs(positions[i]);
+    if (lat > maxLat) {
+      maxLat = lat;
+      pointIndex = i - 1;
+    }
+  }
+  return pointIndex;
+}
+
+function wrapLongitudesForShortestPath(positions, size, startIndex, endIndex) {
+  let prevLng = positions[0];
+  let lng;
+  for (let i = startIndex; i < endIndex; i += size) {
+    lng = positions[i];
+    const delta = lng - prevLng;
+    if (delta > 180 || delta < -180) {
+      lng -= Math.round(delta / 360) * 360;
+    }
+    positions[i] = prevLng = lng;
+  }
+}
+
+function insertPoleVertices(positions, size, startIndex, endIndex) {
+  // Check if the ring contains a pole
+  const firstLng = positions[startIndex];
+  const lastLng = positions[endIndex - size];
+  if (Math.abs(firstLng - lastLng) > 180) {
+    // The ring does not make a round trip
+    // Add the nearest pole to the vertices so that the polygon tesselates correctly
+    const p = getPointAtIndex(positions, 0, size, startIndex);
+    // Copy the first vertex to the world of the last vertex
+    p[0] += Math.round((lastLng - firstLng) / 360) * 360;
+    push(positions, p);
+    // Project the copied vertex to the edge of the map
+    p[1] = Math.sign(p[1]) * MAX_LATITUDE;
+    push(positions, p);
+    // Project the first vertex to the edge of the map
+    p[0] = firstLng;
+    push(positions, p);
+  }
+}
+
+function shiftLongitudesIntoRange(positions, size) {
+  let refLng;
+  const n = positions.length;
```

Comment:
Please don't...  

I could buy `const positionCount = positions.length / size;`, that could make things more readable.